src/observed.decorator.ts

```diff
@@ -97,3 +82,25 @@ export const Observed = (options: ObservedDecoratorOptions = { type: 'behavior'
         });
     };
 };
+
+namespace ObservedDecorator {
+    export function initializeSubject(firstValue: any, options: ObservedDecoratorType | ObservedDecoratorOptions): Subject<any> {
+        switch (getType(options)) {
+            case 'behavior':
+                return new BehaviorSubject(firstValue ?? null);
+            case 'subject':
+                return new Subject();
+            case 'replay':
+                const replayOptions = (options as ObservedDecoratorOptions)?.replayOptions;
+                return new ReplaySubject(replayOptions?.bufferSize, replayOptions?.windowTime, replayOptions?.scheduler);
+            default:
+                return new BehaviorSubject(firstValue ?? null);
+        }
+    }
+
+    export function getType(options: ObservedDecoratorType | ObservedDecoratorOptions): ObservedDecoratorType {
+        return (typeof options === 'string')
+            ? options
+            : options?.type;
+    }
```

Comment:
Since this `getType` function is only used internally here, you don't need to `export` it.

And do we really need a function for `getType` here? Maybe if we end up needing it in other places but now it just feels like a waste of lines

```suggestion
    export function initializeSubject(firstValue: any, options: ObservedDecoratorType | ObservedDecoratorOptions): Subject<any> {
        const type = typeof options === 'string' ? options : options?.type;
        switch (type) {
            case 'behavior':
                return new BehaviorSubject(firstValue ?? null);
            case 'subject':
                return new Subject();
            case 'replay':
                const replayOptions = (options as ObservedDecoratorOptions)?.replayOptions;
                return new ReplaySubject(replayOptions?.bufferSize, replayOptions?.windowTime, replayOptions?.scheduler);
            default:
                return new BehaviorSubject(firstValue ?? null);
        }
    }
```

Reply:
updated.

Tested locally and the type inference on `const type` does indeed work as expected.